auth-manager.js

```diff
@@ -0,0 +1,70 @@
+const mongoose = require('mongoose');
+mongoose.set('useNewUrlParser', true);
+mongoose.set('useFindAndModify', false);
+mongoose.set('useCreateIndex', true);
+mongoose.set('useUnifiedTopology', true);
+
+const bcrypt = require('bcryptjs');
+
+const userAccountSchema = require('./userAccount');
+
+module.exports = function() {
+    var UserAccount;
+
+    return {
+        initialize: function(mongoDBConnectionString) {
+            return new Promise((resolve, reject) =>  {
+                let db = mongoose.createConnection(mongoDBConnectionString);
+                db.on('error', err => reject(err));
+                db.once('open', _ => {
+                    UserAccount = db.model('userAccount', userAccountSchema, 'userAccount');
+                    resolve(db);
+                });
+            });
+        },
+
+        register: async function(newUser) {
+            if (!newUser.userName || newUser.userName == '') {
```

Comment:
Use underscores as separators instead of camel case. It makes it extremely difficult to create clients in mobile

Reply:
And do this change also. The rest of the changes can be done later. I will open new issues for that.